### DISCUSSION TITLE
请教关于FJsEnvImpl::TryBindJs中的判断逻辑

### DISCUSSION THREAD

**Discussion by watsonsong** [Tencent/puerts#1848] [Q&A] [answered]
在FJsEnvImpl::TryBindJs中有如下判断逻辑：
```
const bool IsCDO = HasAnyFlags(RF_ClassDefaultObject | RF_ArchetypeObject);
UClass* Class = InObject->GetClass();
auto TypeScriptGeneratedClass = Cast<UTypeScriptGeneratedClass>(Class);
if (UNLIKELY(TypeScriptGeneratedClass))
{
    if (UNLIKELY(IsCDO))
    {
    }
}
else if (UNLIKELY(!IsCDO && Class == UTypeScriptGeneratedClass::StaticClass()))
{
}
```
这里的逻辑机制没太看明白。如果Cast<UTypeScriptGeneratedClass>(Class)失败，TypeScriptGeneratedClass为空的话，Class == UTypeScriptGeneratedClass::StaticClass()这一句肯定不会相等的吧。但后面那个else分支还能进去，是否这里有什么UE的机制。

另外想了解一下这几个分支分别对应的逻辑情况，万分感谢~~

**Comment by chexiongsheng** [comment, marked as answer]
不是什么UE的机制，是你理解有误。
Cast失败表示Class不是UTypeScriptGeneratedClass的实例。
Class == UTypeScriptGeneratedClass::StaticClass表示InObject是UTypeScriptGeneratedClass的实例。

**Reply by watsonsong** [reply]
明白了，这里是Class的Class类型。这里TryBindJs的几个分支分别代表的是什么情况呢，分别是蓝图类，CDO和实例么？
